ETL/firebase_migration_v2.py

```
@@ -0,0 +1,1921 @@
+		
+		# Fallback: If zip_code is still empty, use ZIP from data
+		if not zip_code:
+			zip_code = str(zip_in_data) if zip_in_data else ""
+
+		end_date = row.get("EndDate", "")
+		if not end_date or not str(end_date).strip():
+			end_date = "12/31/2026"
+		else:
+			# If end_date is present but not a valid date, set to 12/31/2026
+			try:
+				_ = datetime.strptime(end_date.strip(), "%m/%d/%Y")
```

Ruff (0.15.20)

[error] Local variable `datetime` referenced before assignment

(F823)